File: examples/cpp/producer_consumer/Consumer.cpp
#include <stdio.h>
#include "Dragonfly.h"
#include "message_defs.h"


int main( int argc, char *argv[])
{
	try 
	{
		Dragonfly_Module mod( MID_CONSUMER, 0);
		mod.ConnectToMMM();
		mod.Subscribe( MT_TEST_DATA);
		
        std::cout << "Consumer running...\n" << std::endl;
        
		while( 1) 
		{
            CMessage M;
			mod.ReadMessage( &M);
			std::cout << "Received message " << M.msg_type << std::endl;
            
			switch( M.msg_type) {
				case MT_TEST_DATA:
					MDF_TEST_DATA data;
					M.GetData( &data);
					std::cout << "  Data = [a: " << data.a << ", b: " << data.b << ", x: " << data.x << "]" << std::endl;
					break;
			}
		}
	}
	catch( UPipeClosedException &e)
	{
		MyCString s;
		e.AppendTraceToString( s);
		std::cout << "UPipeClosedException: " << s.GetContent() << std::endl;
	}
	catch( UPipeException &e)
	{
		MyCString s;
		e.AppendTraceToString( s);
		std::cout << "UPipeException: " << s.GetContent() << std::endl;
	}
	catch( MyCException &e)
	{
		MyCString s;
		e.AppendTraceToString( s);
		std::cout << "MyCException: " << s.GetContent() << std::endl;
	}
	catch(...)
	{
		MyCString s;
		std::cout << "Unknown Exception!" << std::endl;
	}
	
	std::cout << "Exiting cleanly." << std::endl;
}

File: examples/python/producer_consumer/Consumer.py
#!/usr/bin/python
import time
import PyDragonfly
from PyDragonfly import copy_from_msg
import message_defs as mdefs
import sys

MID_CONSUMER = 11

if __name__ == "__main__":
    mod = PyDragonfly.Dragonfly_Module(MID_CONSUMER, 0)
    mod.ConnectToMMM("localhost:7111")
    mod.Subscribe(mdefs.MT_TEST_DATA)
    
    print "Consumer running...\n"
    
    while (1):
        msg = PyDragonfly.CMessage()
        mod.ReadMessage(msg)    # blocking read
        print "Received message ", msg.GetHeader().msg_type

        if msg.GetHeader().msg_type == mdefs.MT_TEST_DATA:
            msg_data = mdefs.MDF_TEST_DATA()
            copy_from_msg(msg_data, msg)
            print "  Data = [a: %d, b: %d, x: %f]" % (msg_data.a, msg_data.b, msg_data.x)
        
    mod.DisconnectFromMMM()
        
